### PR TITLE
Product master checkbox

### PR DESCRIPTION
**Pull request by omarZrayka23**
checked and textfield overlap

### CODE REVIEW

**Comment by NoraBitarSM** [comment]
try to switch the added checkbox with a row selection handler

**Comment by NoraBitarSM** [comment]
functionality tested, approved

File: src/components/Shared/DataGrid/components/checkbox/edit.js [hunk 1]
```diff
@@ -1,27 +1,22 @@
 // import Checkbox from '@material-ui/core/Checkbox';
 import { Checkbox } from '@mui/material'
-import { useGridApiContext } from '@mui/x-data-grid'
-
-export default function TextFieldEdit({ id, field, value, update }) {
-  const api = useGridApiContext()
 
+export default function TextFieldEdit({ id, field, value, update, row }) {
```
Comment (hadichahine) [inline src/components/Shared/DataGrid/components/checkbox/edit.js:4]: Rename to checkbox edit.

File: src/components/Shared/DataGrid/index.js [hunk 2]
```diff
@@ -23,6 +23,8 @@ export function DataGrid({
   onChange,
   allowDelete = true,
   allowAddNewLine = true,
+  selectedRow,
+  setSelectRow,
```
Comment (hadichahine) [inline src/components/Shared/DataGrid/index.js:27]: Change this to onSelectionChange.

File: src/components/Shared/DataGrid/index.js [hunk 3]
```diff
@@ -23,6 +23,8 @@ export function DataGrid({
   onChange,
   allowDelete = true,
   allowAddNewLine = true,
+  selectedRow,
```
Comment (hadichahine) [inline src/components/Shared/DataGrid/index.js:26]: Why are we passing this?

**Review by hadichahine — CHANGES_REQUESTED** [review]

**Pull request merged**